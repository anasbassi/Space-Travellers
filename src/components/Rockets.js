import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rocket from './Rocket';
import { fetchRocketApi } from '../redux/rockets/rocket';
import '../styles/rockets.css';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRocketApi());
  }, [dispatch]);

  return (
    <div className="rockets-container">
      {rockets.map((rocket) => (
        <Rocket
          key={rocket.id}
          id={rocket.id}
          name={rocket.name}
          description={rocket.description}
          image={rocket.image}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
};

export default Rockets;
