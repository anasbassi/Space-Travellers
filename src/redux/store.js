import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocket';
import checkReducer from './missions/mission';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: checkReducer,
  },
});

export default store;
