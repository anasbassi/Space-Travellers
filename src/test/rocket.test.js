import {
  render, screen, fireEvent, act,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../components/Rockets';
import store from '../redux/store';

describe('Rockets page component', () => {
  afterEach(() => {
    act(() => store.dispatch({
      type: 'rockets/SHOW_ROCKETS',
      payload: [],
    }));
  });

  it('Should reserve a rocket', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    const button = await screen.findAllByText('Reserve Rocket');
    fireEvent.click(button[0]);

    const reserveBadge = await screen.findAllByText('Reserved');
    const cancelButton = await screen.findAllByText('Cancel Reservation');
    expect(reserveBadge.length).toBe(1);
    expect(cancelButton.length).toBe(1);
  });

  it('Should cancel reservation to rocket', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    const reserveButton = await screen.findAllByText('Reserve Rocket');
    fireEvent.click(reserveButton[0]);

    const cancelButton = await screen.findAllByText('Cancel Reservation');
    fireEvent.click(cancelButton[0]);

    // const button = await screen.findAllByText('Reserve Rocket');
    // expect(button.length).toBe(4);
  });

  it('Should match the snapshot', async () => {
    const dom = render(<Provider store={store}><Rockets /></Provider>);
    await act(() => expect(dom).toMatchSnapshot());
  });
});
