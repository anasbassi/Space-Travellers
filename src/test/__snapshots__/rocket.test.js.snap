// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets page component Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rockets-container"
      >
        <div
          class="rocket-container"
          id="1"
        >
          <img
            alt="Rocket-Img"
            src="https://imgur.com/DaCfMsj.jpg"
          />
          <div
            class="rocket-details"
          >
            <h2>
              Falcon 1
            </h2>
            <p>
              <span />
              The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
            </p>
            <button
              class="reserve"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
        <div
          class="rocket-container"
          id="2"
        >
          <img
            alt="Rocket-Img"
            src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
          />
          <div
            class="rocket-details"
          >
            <h2>
              Falcon 9
            </h2>
            <p>
              <span
                class="badge"
              >
                Reserved
              </span>
              Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
            </p>
            <button
              class="cancel"
              type="button"
            >
              Cancel Reservation
            </button>
          </div>
        </div>
        <div
          class="rocket-container"
          id="3"
        >
          <img
            alt="Rocket-Img"
            src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
          />
          <div
            class="rocket-details"
          >
            <h2>
              Falcon Heavy
            </h2>
            <p>
              <span />
              With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
            </p>
            <button
              class="reserve"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
        <div
          class="rocket-container"
          id="4"
        >
          <img
            alt="Rocket-Img"
            src="https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg"
          />
          <div
            class="rocket-details"
          >
            <h2>
              Starship
            </h2>
            <p>
              <span />
              Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
            </p>
            <button
              class="reserve"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rockets-container"
    >
      <div
        class="rocket-container"
        id="1"
      >
        <img
          alt="Rocket-Img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="rocket-details"
        >
          <h2>
            Falcon 1
          </h2>
          <p>
            <span />
            The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
          </p>
          <button
            class="reserve"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
      <div
        class="rocket-container"
        id="2"
      >
        <img
          alt="Rocket-Img"
          src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
        />
        <div
          class="rocket-details"
        >
          <h2>
            Falcon 9
          </h2>
          <p>
            <span
              class="badge"
            >
              Reserved
            </span>
            Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
          </p>
          <button
            class="cancel"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </div>
      <div
        class="rocket-container"
        id="3"
      >
        <img
          alt="Rocket-Img"
          src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
        />
        <div
          class="rocket-details"
        >
          <h2>
            Falcon Heavy
          </h2>
          <p>
            <span />
            With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
          </p>
          <button
            class="reserve"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
      <div
        class="rocket-container"
        id="4"
      >
        <img
          alt="Rocket-Img"
          src="https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg"
        />
        <div
          class="rocket-details"
        >
          <h2>
            Starship
          </h2>
          <p>
            <span />
            Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
          </p>
          <button
            class="reserve"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
